app/js/lib/validation_dispatchers/base.js

```diff
@@ -36,13 +37,31 @@ var _ = require( 'underscore' ),
 				return;
 			}
 
+			if ( !this.fieldsRequiredForDispatchAreSet( selectedValues ) ) {
+				return;
+			}
+
 			this.previousFieldValues = selectedValues;
 
 			if ( this.beginActionCreationFunction ) {
 				store.dispatch( this.beginActionCreationFunction( selectedValues ) );
 			}
 			validationResult = this.validationFunction( selectedValues );
 			return store.dispatch( this.finishActionCreationFunction( validationResult ) );
+		},
+
+		fieldsRequiredForDispatchAreSet: function ( formValues ) {
+			var allFieldsSet = true;
+
+			if ( _.isObject( formValues ) && this.fieldValuesIfNotSet !== null ) {
+				_.each( this.fieldValuesIfNotSet, function ( initalValue, fieldName ) {
+					if ( initalValue === formValues[ fieldName ] ) {
+						allFieldsSet = false;
+					}
+				} );
+			}
+
+			return allFieldsSet;
```

Comment:
Oh oops... I had not seen the `function` :) Still, the imperativeness of this code seriously irks me. I don't know this library (or JS well for that matter), so dunno if there is a better way. @gbirke any idea?